server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1700,11 +1766,12 @@ public void trimUnreferencedReaders() throws IOException {
         } catch (final Exception ex) {
             closeOnTragicEvent(ex);
             throw ex;
+        } finally {
+            writeLock.unlock();
```

Comment:
does this require double try? or only with read lock

Reply:
Only for the read-lock, the locks themselves are properly reentrant and we only acquire the write lock in `close OnTragicEvent`.